desktop/main.dev.ts

```diff
@@ -96,24 +96,22 @@ const handleClosingApp = async () => {
     } else if (response === 1) {
       showCloseAppModal(mainWindow, isDarkMode);
       await nodeManager.stopNode();
-      mainWindow.destroy();
-      app.quit();
     }
   } else {
     const isRunningLocalNode = StoreService.get('localNode');
     if (isRunningLocalNode) {
       showCloseAppModal(mainWindow, isDarkMode);
       await nodeManager.stopNode();
     }
-    mainWindow.destroy();
-    app.quit();
   }
 };
 
-app.on('window-all-closed', () => {
-  // Respect the OSX convention of having the application in memory even
-  // after all windows have been closed
-  if (process.platform !== 'darwin') {
+let closing = false;
+app.on('before-quit', async (event) => {
+  if (!closing) {
+    event.preventDefault();
+    await handleClosingApp();
```

Comment:
handleClosingApp has option to leave app running and minimized to tray
next line will cause app to quit no matter what user chooses to do

Reply:
To clarify, we want the user to confirm quitting the app if smeshing is currently enabled and advise him to minimize smapp instead,